x-pack/plugins/security_solution/server/endpoint/endpoint_app_context_services.ts

```diff
@@ -162,21 +160,8 @@ export class EndpointAppContextService {
   public async getEndpointAuthz(request: KibanaRequest): Promise<EndpointAuthz> {
     const fleetAuthz = await this.getFleetAuthzService().fromRequest(request);
     const userRoles = this.security?.authc.getCurrentUser(request)?.roles ?? [];
-    const { endpointRbacEnabled, endpointRbacV1Enabled } = this.experimentalFeatures;
-    const isPlatinumPlus = this.getLicenseService().isPlatinumPlus();
-    const listClient = this.getExceptionListsClient();
-
-    const hasExceptionsListItems = !isPlatinumPlus
-      ? await doesArtifactHaveData(listClient, ENDPOINT_HOST_ISOLATION_EXCEPTIONS_LIST_ID)
-      : true;
-
-    return calculateEndpointAuthz(
-      this.getLicenseService(),
-      fleetAuthz,
-      userRoles,
-      endpointRbacEnabled || endpointRbacV1Enabled,
-      hasExceptionsListItems
-    );
+
+    return calculateEndpointAuthz(this.getLicenseService(), fleetAuthz, userRoles);
```

Comment:
Why was the `hasExceptionsListItems` property (which makes an api call) removed from here?

My guess is because you changed `calculateEndpointAuthz()` to no longer take in `hasHostIsolationExceptionsItems` argument. Does this mean that the API is not protected now and could allow access even if they are not suppose to (downgrade scenario)?

Reply:
Hey, the data check has been moved to the links.ts, to remove the Host Isolation Exceptions link from the app.

This change is not related to the appFeatures implementation, it can be rolled back without problems, but I think we should consider extracting this check for the following reasons:

- Performance: We are doing this API call many times, even when the role capabilities conditions make this check irrelevant, we need to always pass this flag to the `calculateEndpointAuthz`.

- Centralization: We are checking the license before doing this API call (I guess that was done to improve performance a bit) but this means we are delegating some authorization logic to the implementations, making them know license conditions outside the "authz" module, creating coupling and losing centralization.

- Consistency: This flag is used to grant read authorization without a license when there is data to read, it makes sense for the downgrade scenario, but that means we are "de facto" allowing reading without a license. The only scenario in which the API will return an error is when there is no data to read, which is a bit inconsistent. The only change I am introducing is the API will return an empty array instead of an error when there's no data. I think this way is more consistent and it does not represent any issue compared with what we have now.

I think extracting the `hasHostIsolationExceptionsItems` from the `calculateEndpointAuthz` has some benefits and does not represent any problem.
The data check is still done in the links.ts to make the Host Isolation Exceptions page inaccessible if needed, the same way it works now: 

https://github.com/elastic/kibana/blob/4d9f0c3a6f0c926a733b0fc113dfc20fe1223e07/x-pack/plugins/security_solution/public/management/links.ts#L257-L262

So the UI components inside the page are already gated by the check. 

And if there's any API route that we should not allow reading if the user does not have the proper license/productType, (exporting maybe) we can use the new `canAccessHostIsolationExceptions` flag, which is the same read role capability, but restricted by license (non-serverless) or productType (serverless).

Am I missing something?